tests/mapreduce/test.persisted.js

```diff
@@ -670,6 +670,23 @@ function tests(suiteName, dbName, dbType) {
       });
     });
 
+    it('should throw when user supplies map func unallowed by strict mode (#6234)', function () {
+      var db = new PouchDB(dbName);
+      var doc = {
+        _id: '_design/barbar',
+        views: {
+          scores: {
+            map: "function() { globalVar = 'hello'; })()"
+          }
+        }
+      };
+      return db.post(doc).then(function (success, failure) {
+        should.not.exist(success);
+        should.exist(failure);
+      });
```

Comment:
I should have been clearer, sorry about that! What I meant is:

```js
db.query('barbar/scores').then(function () {
  throw new Error('expected an error');
}, function (err) {
  should.exist(err);
});
```

This ensures that if the promise _succeeds_, the test will fail. But if the promise _fails_, the test will succeed.

Also I believe you will have to add a non-design doc as well, because otherwise the view just won't run because there are no non-design documents.